Patch 33

quantum9Innovation opened the pull request:
This PR works to solve some of the issues mentioned in https://github.com/Quantalabs/DiffPriv/discussions/33. Please note that much of the code added in this release is highly experimental (and not quite tested either; hence the drop in code coverage). We hope to expand on this later before it is shipped into production.

---
See commit for more detailed information.

Closes #34

quantum9Innovation committed:
Begins work on the `mech` module
This is in line with the DiffPriv v2 roadmap (published in
discussion #33).

quantum9Innovation commented:
Will get working on the code coverage issues as soon as possible. However, after the review process is done and if possible, this should be merged right into the v2 branch (after all, it is just an alpha release). @Quantalabs you may need to use admin privileges to accomplish this.

Quantalabs commented on DiffPriv/mech/__init__.py:
```diff
@@ -0,0 +1,8 @@
+from .main import *
+from . import validate
+from . import pre
+from . import utils
```
Why not just move `main` here?

Quantalabs commented on DiffPriv/mech/validate.py:
```diff
@@ -0,0 +1,152 @@
+"""Validates mechanism params"""
+
+class DomainError(Exception):
```
These errors should be in `DiffPriv/utils.py`.

quantum9Innovation replied:
Doing this first (1)

quantum9Innovation replied:
Committing now ...

Quantalabs commented on setup.py:
```diff
@@ -5,7 +5,7 @@
 setuptools.setup(
     name="DiffPriv",
-    version="v1.0.3",
+    version="v2.0.0",
```
```suggestion
    version="v2.0.0-alpha1",
```

Quantalabs replied:
Or:
```suggestion
    version="v2.0.0a1",
```

Quantalabs reviewed (changes requested):
Before merge:
- Regenerate documentation, and check all the docstrings are accurate- make adjustments as needed
- Fix suggestions listed below.

quantum9Innovation committed:
Update setup.py

Co-authored-by: QLabs <55121845+Quantalabs@users.noreply.github.com>

quantum9Innovation commented:
> Regenerate documentation, and check all the docstrings are accurate- make adjustments as needed

Doing this second (2)

Quantalabs commented:
Why are you merging to main? Merge to https://github.com/Quantalabs/DiffPriv/tree/v2

quantum9Innovation commented:
ETA for this PR is ~24 hours
Changes should be made around the same time tomorrow; after merging expect another week before code coverage and testing updates

Quantalabs commented:
> Why are you merging to main? Merge to `v2`

Fixed

Quantalabs commented:
@quantum9Innovation additionally, add tests for classes and functions. If any additional deepsource errors come up, fix those too.

quantum9Innovation commented:
On it (3)

quantum9Innovation commented:
## **Progress**

- [x] (1)
- [x] (2)
- [x] (3)

quantum9Innovation commented:
Unfortunately delays have pushed back the merge date to today. PR should be ready to merge in a few hours.

quantum9Innovation committed:
Fix (1)

Moves error classes to mech/utils.py

Additional Notes:
For full discussion, see:
https://github.com/Quantalabs/DiffPriv/pull/35#discussion_r711828220

quantum9Innovation committed:
Fix (2)

Fix documentation errors/problems after initial generation.

Additional Notes:
For full discussion, see:
https://github.com/Quantalabs/DiffPriv/pull/35#issuecomment-922566814

quantum9Innovation commented:
> Before merge:
> 
> * Regenerate documentation, and check all the docstrings are accurate- make adjustments as needed
> * Fix suggestions listed below.

Done, although it seems accomplishing (3) is harder than it seems (and thus may take some more time). This should likely be postponed until the beta release. @Quantalabs any thoughts?

quantum9Innovation committed:
Fix (2)

Fix documentation errors/problems after initial generation.

Additional Notes:
For full discussion, see:
https://github.com/Quantalabs/DiffPriv/pull/35#issuecomment-922566814

quantum9Innovation committed:
Partially fix (3)

Implement partial testing support for `DiffPriv.mech.Mechanism`.

Additional Notes:
For full discussion, see:
https://github.com/Quantalabs/DiffPriv/pull/35#issuecomment-922569494

quantum9Innovation committed:
Merge branch 'patch-33' of https://github.com/Quantalabs/DiffPriv into patch-33

quantum9Innovation commented:
> @quantum9Innovation additionally, add tests for classes and functions. If any additional deepsource errors come up, fix those too.

All 2 suggestions have been implemented, (3) is complete, and DeepSource is continuing to pass. Ready for merge after review from @Quantalabs
